src_c/surface.c

```diff
@@ -3718,17 +3718,24 @@ pgSurface_Blit(pgSurfaceObject *dstobj, pgSurfaceObject *srcobj,
     }
     else if (the_args != PYGAME_BLEND_ALPHA_SDL2 &&
              !(pg_EnvShouldBlendAlphaSDL2()) &&
-             SDL_GetColorKey(src, &key) != 0 &&
-             (dst->format->BytesPerPixel == 4 ||
-              dst->format->BytesPerPixel == 2) &&
-             _PgSurface_SrcAlpha(src) &&
-             (SDL_ISPIXELFORMAT_ALPHA(src->format->format)) &&
-             !pg_HasSurfaceRLE(src) && !pg_HasSurfaceRLE(dst) &&
-             !(src->flags & SDL_RLEACCEL) && !(dst->flags & SDL_RLEACCEL)) {
-        /* If we have a 32bit source surface with per pixel alpha
-           and no RLE we'll use pygame_Blit so we can mimic how SDL1
-            behaved */
-        result = pygame_Blit(src, srcrect, dst, dstrect, the_args);
+             SDL_GetColorKey(src, &key) != 0 && !pg_HasSurfaceRLE(src) &&
+             !pg_HasSurfaceRLE(dst) && !(src->flags & SDL_RLEACCEL) &&
+             !(dst->flags & SDL_RLEACCEL)) {
+        if ((dst->format->BytesPerPixel == 4 ||
+             dst->format->BytesPerPixel == 2) &&
+            _PgSurface_SrcAlpha(src) &&
+            SDL_ISPIXELFORMAT_ALPHA(src->format->format)) {
+            /* If we have a 32bit or 16 source surface with per pixel alpha
+               and no RLE we'll use pygame_Blit so we can mimic how SDL1
+               behaved */
+            result = pygame_Blit(src, srcrect, dst, dstrect, the_args);
+        }
+        else if (dst->format->BytesPerPixel == src->format->BytesPerPixel) {
```

Comment:
It's not that the bit depth needs to be the same, the pixelformat needs to be the same.

Like if you had a BGRX surface blitting onto a RGBX surface, the correct move is not a straight copy.

Reply:
Good point. Don't expect this comes up that often but it would need to be handled here.

Reply:
I suppose my suggestion would be

`else if (dst->format->format == src->format->format && !SDL_ISPIXELFORMAT_ALPHA(dst->format->format) && !SDL_ISPIXELFORMAT_INDEXED(dst->format->format) {}`

Because they need to have the same pixelformat, they should not have alpha, and it should not be palettized.

Reply:
I guess we need some more tests for the pixel format cases, and wouldn't be surprised if those tests uncovered existing bugs too.